### ISSUE TITLE
[QUESTION] Can `Object` properties should mapped into `Map<>`?

### ISSUE THREAD

**Issue by fncap** [epam-cross-platform-lab/swagger-dart-code-generator#296]
**Please describe a problem.**
Cannot retrieve values from _Dictionary like_ properties, as per generated `Object?` type instead
**Describe the solution you'd like**
Some times, data returned from WebApi, has form of Dictionary like structures.
Swagger recognise them as a generic `object` type, and your tool do the same, generating `Object?` typed properties.
Instead of that, should be better if tool treats those properties as a `Map<string,string>` or something like that, in that way programmer should free to read values from Map as he want!

**Comment by fryette** [comment]
But why you cannot transform `Object?` to `Map<String, dynamic>` ?

**Comment by fncap** [comment]
Yes! And this is exactly what I did!
But, for me that I'm new with Dart, it was not clear that could safely cast an Object to a Map<string, dynamic> type!
Thanks for your time!

**Comment by fryette** [comment]
@fncap anyway we will try to fix it(check related issue) to provide the best experience with the library.

Thanks and have a good weekends!

**Comment by Vovanella95** [comment]
Hi @fncap 

We investigated this issue. We can't map Object properties to `Map<String, String>`, because `metadata` can has properties + additional properties. So mapping to Map<String, any concrete type> is not possible.

The only one way is to map it to Map<String, dynamic>. Is it fine?

**Comment by fncap** [comment]
Yes you are right, generating a 'Map<string,dynamic>' should be fine!

**Comment by Vovanella95** [comment]
Hi @fncap 

Added generation of Map<String, dynamic>? for such fields
